base/iostream.jl

```diff
@@ -225,8 +225,18 @@ function readuntil(s::IOStream, delim::UInt8)
     ccall(:jl_readuntil, Array{UInt8,1}, (Ptr{Void}, UInt8, UInt8), s.ios, delim, 0)
 end
 
-function readline(s::IOStream)
-    ccall(:jl_readuntil, Ref{String}, (Ptr{Void}, UInt8, UInt8), s.ios, '\n', 1)
+function readline(s::IOStream, chomp::Bool=false)
+    chomp || return ccall(:jl_readuntil, Ref{String}, (Ptr{Void}, UInt8, UInt8), s.ios, '\n', 1)
+
+    s = ccall(:jl_readuntil, Ref{String}, (Ptr{Void}, UInt8, UInt8), s.ios, '\n', 1)
+    i = endof(s)
+    if i < 1 || codeunit(s,i) != 0x0a
+        return s[1:i]
```

Comment:
For this PR at least, we don't need these functions to be exported since we're in Base. But here we would need `jl_readuntil` to return a `StringVector`. Is that possible?